File: js/13335.js
const fs = require('fs');

const inputs = fs.readFileSync('example.txt').toString().trim().split('\n');

const [n, L, w] = inputs[0].split(' ').map(Number);
const truckWeights = inputs[1].split(' ').map(Number);
const q = [];
let idx = 0;
let t = 0;
let currW = 0;

while (true) {
  if (idx === n) {
    t += L;
    break;
  }

  t++;

  if (q.length === L) {
    currW -= q.shift();
  }

  const tw = truckWeights[idx];

  if (currW + tw <= w) {
    currW += tw;
    q.push(tw);
    idx++;
  } else {
    q.push(0);
  }
}

console.log(t);

File: c++/13335.cpp
#include <iostream>
#include <queue>
#include <vector>

using namespace std;

queue<int> q;
int ans, idx, currW;

int main() {
  freopen("example.txt", "r", stdin);

  int n, w, L;
  cin >> n >> L >> w;

  vector<int> truckWeights;
  for (int i = 0; i < n; i++) {
    int a;
    cin >> a;
    truckWeights.push_back(a);
  }

  while (1) {
    if (idx == n) {
      ans += L;
      break;
    }

    ans++;

    if (q.size() == L) {
      currW -= q.front();
      q.pop();
    }

    int tw = truckWeights[idx];
    if (currW + tw <= w) {
      currW += tw;
      q.push(tw);
      idx++;
    } else {
      q.push(0);
    }
  }
  cout << ans;

  return 0;
}